js/game.js

```diff
@@ -0,0 +1,253 @@
+import {Player} from './players.js';
+import {Board} from './board.js';
+
+function returnAccumulatedScores(data) {
+
+  let dataToAccumulate = [];
+
+  do {
+
+    let elementToCompare = data.shift();
+    elementToCompare.apperances = 0;
+    for (let i = 0; i < data.length; i++) {
+      if (elementToCompare.x === data[i].x && elementToCompare.y === data[i].y) {
+        elementToCompare.score += data[i].score;
+        elementToCompare.apperances++;
+        data.splice(i, 1);
+        i--;
+      }
+    }
+    dataToAccumulate = [
+      ...dataToAccumulate,
+      elementToCompare
+    ];
+
+  } while (data.length > 0)
+
+  const dataToReturn = dataToAccumulate.map(element => {
+    element.evaluate = (element.score / element.apperances)
+    return element;
+  });
+
+  return dataToReturn
+}
+
+class Game {
+  constructor() {
+    this.board = new Board;
+    this.players = this.createPlayers()
+    this.ready = false;
+    this.fieldsLeft = 9;
+  }
+
+  createPlayers() {
+    return [
+      new Player('Player 1', 'cross', true),
+      new Player('Player 2', 'circle')
+    ];
+
+  }
+
+  get activePlayer() {
+    return this.players.find(player => player.active);
+  }
+
+  get unactivePlayer() {
+    return this.players.find(player => !player.active);
+  }
+
+  switchPlayers() {
+    this.players.forEach(player => player.active = !player.active);
+  }
+
+  startGame(boardDiv) {
+    this.board.renderFields(boardDiv);
+    this.ready = true;
+  }
+
+  makeMove(x,y){
+    const element = document.querySelector(`[data-x="${x}"][data-y="${y}"]`);
+    const field = this.board.fields[x][y];
+    field.free = false;
+    field.owner = this.activePlayer;
+    field.renderSymbol(element);
+    this.fieldsLeft--;

```

Comment:
Tak btw. to środowisko `JSowe` jest podzielone jeśli chodzi o inkrementację i dekrementację robioną w ten sposób. W wielu innych językach uważa się, że taki zapis za złą praktykę i w `JSie` to trochę szara strefa.

Dyskusja na ten temat: https://stackoverflow.com/questions/971312/why-avoid-increment-and-decrement-operators-in-javascript

Nie wiem jak podejdzie do tego ktoś kto będzie patrzył na Twój kod więc stosowałbym zapisy:

`this.fieldsetLeft -= 1`

Reply:
ok, nie mialem pojecia